outbox.js

```diff
@@ -0,0 +1,205 @@
+class DistributedOutbox extends HTMLElement {
+  constructor() {
+    super();
+    this.renderedItems = new Set(); // Tracks rendered items
+    this.numPosts = 1; // Default value
+    this.page = 1; // Default value
+    this.totalPages = 0; // Keep track of total pages
+  }
+
+  static get observedAttributes() {
+    return ["url", "num-posts", "page"];
+  }
+
+  connectedCallback() {
+    // Use attributes or default values
+    this.numPosts = parseInt(this.getAttribute("num-posts")) || this.numPosts;
```

Comment:
```suggestion
    this.numPosts = parseInt(this.getAttribute("num-posts"), 10) || this.numPosts;
```

Generally it's good practice to supply a radix for `parseInt`.